scripts/flitcli/flit_bisect.py

```diff
@@ -451,16 +451,32 @@ def bisect_search(is_bad, elements):
     >>> call_count
     9
 
+    Test out the found_callback() functionality.
+    >>> s = set()
+    >>> y = bisect_search(is_bad, [-1, -2, -3, -4], found_callback=s.add)
+    >>> sorted(y)
+    [-4, -3, -2, -1]
+    >>> sorted(s)
+    [-4, -3, -2, -1]
+
     See what happens when it has a pair that only show up together and not
     alone.  Only if -6 and 5 are in the list, then is_bad returns true.
     The assumption of this algorithm is that bad elements are independent,
     so this should throw an exception.
     >>> def is_bad(x,y):
     ...     return max(x) - min(x) > 10
-    >>> x = bisect_search(is_bad, [-6, 2, 3, -3, -1, 0, 0, -5, 5])
+    >>> bisect_search(is_bad, [-6, 2, 3, -3, -1, 0, 0, -5, 5])
     Traceback (most recent call last):
         ...
     AssertionError: Assumption that bad elements are independent was wrong
+
+    Check that the found_callback is not called on false positives.  Here I
+    expect no output since no single element can be found.
+    >>> try:
+    ...     bisect_search(is_bad, [-6, 2, 3, -3, -1, 0, 0, -5, 5],
+    ...                   found_callback=print)
+    ... except AssertionError:
+    ...     pass
```

Comment:
This test that I added fails without the fix I put in.  True, in cases where it would fail, we also would have the assertion failure.  But still, we could get weird results such as a printout into the log that a file was responsible right before the assertion gets thrown.